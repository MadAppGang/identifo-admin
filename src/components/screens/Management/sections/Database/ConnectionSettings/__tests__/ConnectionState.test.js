import { ConnectionState } from '../ConnectionState';
import {
  CONNECTION_ESTABLISHED,
  CONNECTION_FAILED,
  CONNECTION_TEST_REQUIRED,
} from '~/modules/database/connectionReducer';

describe('database <ConnectionState />', () => {
  let props;

  beforeEach(() => {
    props = {
      checking: false,
      loading: false,
      state: CONNECTION_TEST_REQUIRED,
      testConnection: jest.fn(),
    };
  });

  test('matches snapshot when connection test required', () => {
    expect(shallow(<ConnectionState {...props} />)).toMatchSnapshot();
  });

  test('matches snapshot when connection established', () => {
    props.state = CONNECTION_ESTABLISHED;
    expect(shallow(<ConnectionState {...props} />)).toMatchSnapshot();
  });

  test('matches snapshot when connection failed', () => {
    props.state = CONNECTION_FAILED;
    expect(shallow(<ConnectionState {...props} />)).toMatchSnapshot();
  });

  test('calls testConnection on button click when test required', () => {
    shallow(<ConnectionState {...props} />).find('Button').simulate('click');
    expect(props.testConnection).toBeCalled();
  });
});
