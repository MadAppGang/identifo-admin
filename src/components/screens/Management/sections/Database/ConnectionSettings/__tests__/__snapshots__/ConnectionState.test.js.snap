// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`database <ConnectionState /> matches snapshot when connection established 1`] = `
<Fragment>
  <div
    className="iap-db__connection iap-db__connection--established"
  >
    <ConnectionIcon
      className="iap-db__connection-icon"
    />
    Connection Established
  </div>
</Fragment>
`;

exports[`database <ConnectionState /> matches snapshot when connection failed 1`] = `
<Fragment>
  <div
    className="iap-db__connection iap-db__connection--failed"
  >
    <ErrorIcon
      className="iap-db__connection-icon"
    />
    Connection Failed
  </div>
</Fragment>
`;

exports[`database <ConnectionState /> matches snapshot when connection test required 1`] = `
<Fragment>
  <Button
    Icon={[Function]}
    disabled={false}
    error={false}
    iconClassName=""
    onClick={[MockFunction]}
    stretch={false}
    transparent={false}
    type="button"
  >
    Test connection
  </Button>
</Fragment>
`;
